python/cudf/cudf_pandas_tests/test_cudf_pandas.py

```diff
@@ -1566,3 +1566,14 @@ def test_arrow_string_arrays():
     )
 
     tm.assert_equal(cu_arr, pd_arr)
+
+
+@pytest.mark.parametrize("indexer", ["at", "iat"])
+def test_at_iat(indexer):
```

Comment:
Does this cover the case of a new column that doesn't exist? I'd like to see a test that more clearly resembles the patterns in the minimal repro of https://github.com/rapidsai/cudf/issues/16112.

Reply:
Yup, added a unit test in [a43454c](https://github.com/rapidsai/cudf/pull/16177/commits/a43454cf6fa157ef1ca5779cdee587334e5b5f82) to mirror the issue. (It appears that `iat.__setitem__` in pandas doesn't support the type of behavior demonstrated in the issue)

Reply:
Huh, thanks for encouraging a unit test with the issue. Even thought the test passed locally the CI didn't agree.

Left some investigation notes in https://github.com/rapidsai/cudf/issues/16112#issuecomment-2207103425 that I may come back to later